### ISSUE TITLE
Issue with running tests for just astropy.io.misc

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#10280]
I'm running into issues when trying to run just the ``astropy.io.misc`` tests:

```
$ pip install -e .
...
$ pytest astropy/io/misc --sw
====================================================================== test session starts ======================================================================
platform linux -- Python 3.8.2, pytest-5.3.5, py-1.8.1, pluggy-0.13.1

Running tests with Astropy version 4.1.dev1579+g770bd2c5c.d20200503.
Running tests in astropy/io/misc.

Date: 2020-05-03T17:21:58

Platform: Linux-5.4.0-26-generic-x86_64-with-glibc2.29

Executable: /home/tom/python/dev/bin/python3.8

Full Python Version: 
3.8.2 (default, Apr 27 2020, 15:53:34) 
[GCC 9.3.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.16.6
Scipy: 1.4.1
Matplotlib: 3.2.0
h5py: 2.10.0
Pandas: 1.0.1
Cython: not available
Scikit-image: 0.16.2
asdf: 2.6.0

Using Astropy options: remote_data: none.

Matplotlib: 3.2.0
Freetype: 2.6.1
rootdir: /home/tom/Code/astropy, inifile: setup.cfg
plugins: openfiles-0.4.1.dev10+g0203b1b, astropy-header-0.1.2, asdf-2.6.0, arraydiff-0.3, remotedata-0.3.2, mpl-0.11, filter-subpackage-0.1.1, cov-2.8.1, doctestplus-0.5.0, hypothesis-5.8.0
collected 410 items / 10 errors / 400 selected                                                                                                                  
stepwise: no previously failed tests, not skipping.

============================================================================ ERRORS =============================================================================
__________________________________________________________ ERROR collecting astropy/io/misc/connect.py __________________________________________________________
astropy/time/core.py:2592: in update_leap_seconds
    table = iers.LeapSeconds.auto_open(files)
astropy/utils/iers/iers.py:1027: in auto_open
    warn('leap-second file is expired.', IERSStaleWarning)
E   astropy.utils.iers.iers.IERSStaleWarning: leap-second file is expired.

During handling of the above exception, another exception occurred:
astropy/io/misc/connect.py:7: in <module>
    hdf5.register_hdf5()
astropy/io/misc/hdf5.py:390: in register_hdf5
    from astropy.table import Table
astropy/table/__init__.py:56: in <module>
    from .serialize import SerializedColumn, represent_mixins_as_columns
astropy/table/serialize.py:10: in <module>
    from astropy.coordinates import representation as coorep
astropy/coordinates/__init__.py:16: in <module>
    from .builtin_frames import *
astropy/coordinates/builtin_frames/__init__.py:29: in <module>
    from .fk5 import FK5
astropy/coordinates/builtin_frames/fk5.py:8: in <module>
    from astropy.coordinates import earth_orientation as earth
astropy/coordinates/earth_orientation.py:19: in <module>
    jd1950 = Time('B1950').jd
astropy/time/core.py:395: in __new__
    update_leap_seconds()
astropy/time/core.py:2596: in update_leap_seconds
    warn("leap-second auto-update failed due to the following "
E   astropy.utils.exceptions.AstropyWarning: leap-second auto-update failed due to the following exception: IERSStaleWarning('leap-second file is expired.')
___________________________________________________________ ERROR collecting astropy/io/misc/yaml.py ____________________________________________________________
astropy/io/misc/yaml.py:240: in <module>
    for cls in (list(coords.representation.REPRESENTATION_CLASSES.values())
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
______________________________________________________ ERROR collecting astropy/io/misc/asdf/extension.py _______________________________________________________
astropy/io/misc/asdf/extension.py:18: in <module>
    from .tags.coordinates.representation import *
astropy/io/misc/asdf/tags/coordinates/representation.py:8: in <module>
    class RepresentationType(AstropyType):
astropy/io/misc/asdf/tags/coordinates/representation.py:13: in RepresentationType
    _representation_module = astropy.coordinates.representation
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
___________________________________________ ERROR collecting astropy/io/misc/asdf/tags/coordinates/representation.py ____________________________________________
astropy/io/misc/asdf/tags/coordinates/representation.py:8: in <module>
    class RepresentationType(AstropyType):
astropy/io/misc/asdf/tags/coordinates/representation.py:13: in RepresentationType
    _representation_module = astropy.coordinates.representation
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
__________________________________________ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py ___________________________________________
astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py:16: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:18: in <module>
    from .tags.coordinates.representation import *
astropy/io/misc/asdf/tags/coordinates/representation.py:8: in <module>
    class RepresentationType(AstropyType):
astropy/io/misc/asdf/tags/coordinates/representation.py:13: in RepresentationType
    _representation_module = astropy.coordinates.representation
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
__________________________________________ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py ___________________________________________
astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py:16: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:18: in <module>
    from .tags.coordinates.representation import *
astropy/io/misc/asdf/tags/coordinates/representation.py:8: in <module>
    class RepresentationType(AstropyType):
astropy/io/misc/asdf/tags/coordinates/representation.py:13: in RepresentationType
    _representation_module = astropy.coordinates.representation
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
__________________________________________ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py __________________________________________
astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py:14: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:18: in <module>
    from .tags.coordinates.representation import *
astropy/io/misc/asdf/tags/coordinates/representation.py:8: in <module>
    class RepresentationType(AstropyType):
astropy/io/misc/asdf/tags/coordinates/representation.py:13: in RepresentationType
    _representation_module = astropy.coordinates.representation
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
__________________________________________ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py __________________________________________
astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py:14: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:18: in <module>
    from .tags.coordinates.representation import *
astropy/io/misc/asdf/tags/coordinates/representation.py:8: in <module>
    class RepresentationType(AstropyType):
astropy/io/misc/asdf/tags/coordinates/representation.py:13: in RepresentationType
    _representation_module = astropy.coordinates.representation
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
______________________________________________________ ERROR collecting astropy/io/misc/tests/test_yaml.py ______________________________________________________
astropy/io/misc/tests/test_yaml.py:26: in <module>
    from astropy.io.misc.yaml import load, load_all, dump
astropy/io/misc/yaml.py:240: in <module>
    for cls in (list(coords.representation.REPRESENTATION_CLASSES.values())
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
______________________________________________________ ERROR collecting astropy/io/misc/tests/test_yaml.py ______________________________________________________
astropy/io/misc/tests/test_yaml.py:26: in <module>
    from astropy.io.misc.yaml import load, load_all, dump
astropy/io/misc/yaml.py:240: in <module>
    for cls in (list(coords.representation.REPRESENTATION_CLASSES.values())
E   AttributeError: module 'astropy.coordinates' has no attribute 'representation'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 10 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================== 10 errors in 1.38s =======================================================================
```

Somehow this only works properly if astropy.coordinates is imported first, e.g. ``pytest astropy/coordinates astropy/io/misc --sw``

**Comment by taldcroft** [comment]
@astrofrog - this should be fixed by #10282.  I had exactly the same problems with `io.ascii` and confirmed that all is well now.

**Comment by astrofrog** [comment]
Indeed looks like this is fixed by #10282
